matlab/+BalanceLaws/compute_numerical_solution.m

```diff
@@ -126,21 +132,42 @@
 %%
 switch time_integrator_num_fields
     case 2
-        while num_steps_run > RK_block_size
-            kernel_list = repmat(RK_Step, 1, RK_block_size);
-
-            t = cl_run_kernel(I_Tech('device'), kernel_list, I_BalanceLaws('g_range'), I_BalanceLaws('l_range'), ...
-                              field_u1, field_u2, ...
-                              current_time, 0);
-            kernel_runtime =  kernel_runtime + t;
-            num_steps_run = num_steps_run - RK_block_size;
-        end
-        if num_steps_run > 0
-            kernel_list = repmat(RK_Step, 1, num_steps_run);
-            t = cl_run_kernel(I_Tech('device'), kernel_list, I_BalanceLaws('g_range'), I_BalanceLaws('l_range'), ...
-                                          field_u1, field_u2, ...
-                                          current_time, 0);
-            kernel_runtime = kernel_runtime + t;
+        if I_RunOps('save_integrals_over_time')
+            for step = 1:I_TI('num_steps')
+                for comp=0:I_BalanceLaws('NUM_CONSERVED_VARS')-1
+                    components(1) = comp;
+                    cl_run_kernel(I_Tech('device'), 'analytical_u', I_BalanceLaws('g_range'), I_BalanceLaws('l_range'), field_u2, current_time, 0);
+                    norm2_output(:) = 0;
+                    cl_run_kernel(I_Tech('device'), 'norm2_diff', I_Tech('g_range'), I_Tech('l_range'), field_u1, field_u2, norm2_output, components, 0);
+                    if strcmp(I_RunOps('norm'),'L2')
+                        error(comp + 1, step) = sqrt(sum(norm2_output));
+                    elseif strcmp(I_RunOps('norm'),'Linf')
+                        error(comp + 1, step) = max(sqrt(norm2_output));
```

Comment:
I don't remember any $L^\infty$ error calculation in the OpenCL code - did I miss it or is it missing? Based on the answer, we should add use `max(norm2_output)` instead of `max(sqrt(norm2_output))` or we should throw an error until such a kernel has been added.